src/components/tree-detail/last-waterings/watering-card.tsx

```diff
@@ -51,6 +56,67 @@ export const WateringCard: React.FC<WateringCardProps> = ({ wateringData }) => {
 		setIsConfirmDeleteVisible(false);
 	};
 
+	const checkForContactRequestAllowed = async () => {
+		if (isWateringByUser) {
+			return;
+		}
+
+		if (!username) {
+			(
+				document.getElementById("login-first-alert") as HTMLDialogElement
+			).showModal();
+			return;
+		}
+
+		setIsContactRequestLoading(true);
+
+		const data = await supabaseClient.functions.invoke(
+			"check_contact_request",
+			{
+				body: {
+					recipientContactName: wateringData.username,
+				},
+			},
+		);
+		setIsContactRequestLoading(false);
+
+		if (!data || data.error) {
+			(
+				document.getElementById(
+					"generic-contact-error-alert",
+				) as HTMLDialogElement
+			).showModal();
+			return;
+		}
+
+		if (data.data) {
```

Comment:
I'd suggest to switch it around here to prevent nested ifs.
```
if (!data.data) {
  return;
}

setSelectedContactRecipientUsername(wateringData.username);
...
```